### PR TITLE
exp branch: Make `branch` arg optional.

### PR DESCRIPTION
**Pull request by daavoo**

### CODE REVIEW

File: dvc/commands/experiments/branch.py [hunk 1]
```diff
@@ -26,5 +26,13 @@ def add_parser(experiments_subparsers, parent_parser):
...
         "experiment", help="Experiment to be promoted."
     )
-    experiments_branch_parser.add_argument("branch", help="Name for the new Git branch")
+    experiments_branch_parser.add_argument(
+        "branch",
+        nargs="?",
+        default=None,
+        help=(
+            "Optional name for the new Git branch. "
+            "Defaults to the name of 'experiment'."
```
Comment (dberenbaum) [inline dvc/commands/experiments/branch.py:35]:
```suggestion
            "Defaults to 'exp-{experiment-name}'."
```

Comment (dberenbaum) [reply dvc/commands/experiments/branch.py:35]: Let's also remember to update the docs 🙏

**Comment by dberenbaum** [comment]
Tested this out a bit more and having minor second thoughts about naming. Tldr maybe we should name the branch something like `{experiment-name}-branch` to better differentiate it from the experiment.

Here's how it looks in practice:

```bash
$ dvc exp show
 ───────────────────────────────────────────────────────────────────────────────────────>
  Experiment                 Created        train.step.loss   train.step.acc   train.epo>
 ───────────────────────────────────────────────────────────────────────────────────────>
  workspace                  -                            -                -            >
  hydra                      Mar 22, 2023                 -                -            >
  ├── 0362005 [crass-snow]   Mar 23, 2023           0.25659          0.92795            >
  ├── 7776043 [oiled-stir]   Mar 23, 2023           0.23829          0.93242            >
  ├── 73f3dcc [wormy-wart]   Mar 23, 2023           0.23982          0.93194            >
  ├── 3761e4b [baggy-raid]   Mar 23, 2023           0.55053          0.90761            >
  ├── e35978a [moldy-plow]   Mar 23, 2023           0.29972          0.91976            >
  └── f9dbcac [dazed-kobs]   Mar 23, 2023                 -                -            >
 ───────────────────────────────────────────────────────────────────────────────────────>

$ dvc exp branch crass-snow
Git branch 'exp-crass-snow' has been created from experiment 'crass-snow'.
To switch to the new branch run:

        git checkout exp-crass-snow

$ dvc exp show
 ───────────────────────────────────────────────────────────────────────────────────────>
  Experiment                     Created        train.step.loss   train.step.acc   train>
 ───────────────────────────────────────────────────────────────────────────────────────>
  workspace                      -                            -                -        >
  hydra                          Mar 22, 2023                 -                -        >
  ├── 0362005 [exp-crass-snow]   Mar 23, 2023           0.25659          0.92795        >
  ├── 7776043 [oiled-stir]       Mar 23, 2023           0.23829          0.93242        >
  ├── 73f3dcc [wormy-wart]       Mar 23, 2023           0.23982          0.93194        >
  ├── 3761e4b [baggy-raid]       Mar 23, 2023           0.55053          0.90761        >
  ├── e35978a [moldy-plow]       Mar 23, 2023           0.29972          0.91976        >
  └── f9dbcac [dazed-kobs]       Mar 23, 2023                 -                -        >
 ───────────────────────────────────────────────────────────────────────────────────────>

$ git checkout exp-crass-snow

# Make some changes and commit them

$ dvc exp show -n 3
 ───────────────────────────────────────────────────────────────────────────────────────>
  Experiment                 Created        train.step.loss   train.step.acc   train.epo>
 ───────────────────────────────────────────────────────────────────────────────────────>
  workspace                  -                      0.25659          0.92795            >
  exp-crass-snow             07:31 AM               0.25659          0.92795            >
  crass-snow                 Mar 23, 2023           0.25659          0.92795            >
  hydra                      Mar 22, 2023                 -                -            >
  ├── 0362005 [crass-snow]   Mar 23, 2023           0.25659          0.92795            >
  ├── 7776043 [oiled-stir]   Mar 23, 2023           0.23829          0.93242            >
  ├── 73f3dcc [wormy-wart]   Mar 23, 2023           0.23982          0.93194            >
  ├── 3761e4b [baggy-raid]   Mar 23, 2023           0.55053          0.90761            >
  ├── e35978a [moldy-plow]   Mar 23, 2023           0.29972          0.91976            >
  └── f9dbcac [dazed-kobs]   Mar 23, 2023                 -                -            >
 ───────────────────────────────────────────────────────────────────────────────────────>
```

Feels a bit confusing that there is the original experiment `crass-snow` and the branch that now has additional changes called `exp-crass-snow`.

**Comment by daavoo** [comment]
> Tldr maybe we should name the branch something like {experiment-name}-branch to better differentiate it from the experiment.

Don't mean to diminish and I don't have a strong opinion, but I don't really understand the difference or the scenario you showed 😅

**Comment by dberenbaum** [comment]
Sorry, not a blocker, but here's and alternate tldr: once you make other changes to the branch you created, calling it `exp-crass-snow` may be confusing when there's a separate exp ref called `crass-snow`. If we call it `crass-snow-branch`, it's more clear which is the original experiment and which is the branch generated from it.

**Comment by daavoo** [comment]
> Sorry, not a blocker, but here's and alternate tldr: once you make other changes to the branch you created, calling it `exp-crass-snow` may be confusing when there's a separate exp ref called `crass-snow`. If we call it `crass-snow-branch`, it's more clear which is the original experiment and which is the branch generated from it.

Makes sense. Updated
